set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java

```diff
@@ -855,5 +860,37 @@ private void setSerializedItemIds(UserSet userSet) {
 	((WebUserSetImpl) userSet).setSerializedItems(jsonSerialized);
 //	userSet.setItems(null);
     }
+    
+    /*
+     * (non-Javadoc)
+     * 
+     * @see eu.europeana.UserSet.web.service.UserSetService#storeUserSet(eu.
+     * europeana.UserSet.definitions.model.UserSet, boolean)
+     */
+//    @Override
+    public UserSet updateUserSet(PersistentUserSet persistentUserSet, UserSet webUserSet) throws HttpException {
+    mergeUserSetProperties(persistentUserSet, webUserSet);
+    // update modified date
+    persistentUserSet.setModified(new Date());
+    updateTotal(persistentUserSet);
+    checkDuplicateUserSets(persistentUserSet, true);
+    UserSet updatedUserSet =  getMongoPersistence().update(persistentUserSet);
+    getUserSetUtils().updatePagination(updatedUserSet, getConfiguration().getUserSetBaseUrl());
+    return updatedUserSet;
+    
+    }
+    
+    public void checkDuplicateUserSets(UserSet userSet, boolean withoutItself) throws SetUniquenessValidationException {
+      //check the set uniqueness only for the EntityBestItemsSet type
+      if(userSet.getType().compareTo(UserSetTypes.ENTITYBESTITEMSSET.toString())==0) {
```

Comment:
please use the equals or equalsIgnoreCase method and use hte constant as the first term of the comparison

Reply:
Ok.